sycl/plugins/unified_runtime/pi2ur.hpp

```diff
@@ -7,37 +7,48 @@
 //===------------------------------------------------------------------===//
 #pragma once
 
-#include <unordered_map>
+#include <optional>
 
 #include "ur_api.h"
 #include <sycl/detail/pi.h>
 #include <ur/ur.hpp>
 
 // Map of UR error codes to PI error codes
 static pi_result ur2piResult(ur_result_t urResult) {
-  std::unordered_map<ur_result_t, pi_result> ErrorMapping = {
-      {UR_RESULT_SUCCESS, PI_SUCCESS},
-      {UR_RESULT_ERROR_UNKNOWN, PI_ERROR_UNKNOWN},
-      {UR_RESULT_ERROR_DEVICE_LOST, PI_ERROR_DEVICE_NOT_FOUND},
-      {UR_RESULT_ERROR_INVALID_OPERATION, PI_ERROR_INVALID_OPERATION},
-      {UR_RESULT_ERROR_INVALID_PLATFORM, PI_ERROR_INVALID_PLATFORM},
-      {UR_RESULT_ERROR_INVALID_ARGUMENT, PI_ERROR_INVALID_ARG_VALUE},
-      {UR_RESULT_ERROR_INVALID_VALUE, PI_ERROR_INVALID_VALUE},
-      {UR_RESULT_ERROR_INVALID_EVENT, PI_ERROR_INVALID_EVENT},
-      {UR_RESULT_ERROR_INVALID_BINARY, PI_ERROR_INVALID_BINARY},
-      {UR_RESULT_ERROR_INVALID_KERNEL_NAME, PI_ERROR_INVALID_KERNEL_NAME},
-      {UR_RESULT_ERROR_INVALID_FUNCTION_NAME, PI_ERROR_BUILD_PROGRAM_FAILURE},
-      {UR_RESULT_ERROR_INVALID_WORK_GROUP_SIZE,
-       PI_ERROR_INVALID_WORK_GROUP_SIZE},
-      {UR_RESULT_ERROR_MODULE_BUILD_FAILURE, PI_ERROR_BUILD_PROGRAM_FAILURE},
-      {UR_RESULT_ERROR_OUT_OF_DEVICE_MEMORY, PI_ERROR_OUT_OF_RESOURCES},
-      {UR_RESULT_ERROR_OUT_OF_HOST_MEMORY, PI_ERROR_OUT_OF_HOST_MEMORY}};
-
-  auto It = ErrorMapping.find(urResult);
-  if (It == ErrorMapping.end()) {
+  switch (urResult) {
+  case UR_RESULT_SUCCESS:
+    return PI_SUCCESS;

```

Comment:
Maybe just this change (shortcut test for success) is enough? I don't believe other cases are ever in a hotspot.

Reply:
Probably yes - I observed boost when just hacked this method to always return SUCCESS. However, if change which removes all maps is ready, why not take it?